internal/service/advancedcluster/data_source_advanced_cluster.go

```diff
@@ -284,6 +285,12 @@ func dataSourceRead(ctx context.Context, d *schema.ResourceData, meta any) diag.
 	}
 	zoneNameToOldReplicationSpecMeta, err := GetReplicationSpecAttributesFromOldAPI(ctx, projectID, clusterName, connV220240530.ClustersApi)
 	if err != nil {
+		if apiError, ok := admin20240530.AsError(err); ok {
+			if apiError.GetErrorCode() == "ASYMMETRIC_SHARD_UNSUPPORTED" && !useReplicationSpecPerShard {
+				return diag.FromErr(fmt.Errorf("please add `use_replication_spec_per_shard = true` to your data source configuration to enable asymmetric shard support. Refer to documentation for more details. %s", err))
+			}
+			return diag.FromErr(fmt.Errorf(errorRead, clusterName, err))
```

Comment:
I believe there is an additional case we are missing, if the error is ASYMMETRIC_SHARD_UNSUPPORTED but useReplicationSpecPerShard is enabled we should continue without any error.